dropshot/src/server.rs

```diff
@@ -545,18 +562,43 @@ type InnerHttpsServerStarterNewReturn<C> =
 impl<C: ServerContext> InnerHttpsServerStarter<C> {
     /// Begins execution of the underlying Http server.
     fn start(
-        self,
-        close_signal: tokio::sync::oneshot::Receiver<()>,
+        mut self,
+        mut close_signal: tokio::sync::oneshot::Receiver<()>,
         log_close: Logger,
-    ) -> tokio::task::JoinHandle<Result<(), hyper::Error>> {
-        let graceful = self.0.with_graceful_shutdown(async move {
-            close_signal.await.expect(
-                "dropshot server shutting down without invoking close()",
-            );
-            info!(log_close, "received request to begin graceful shutdown");
-        });
+    ) -> tokio::task::JoinHandle<()> {
+        use hyper_util::rt::{TokioExecutor, TokioIo, TokioTimer};
+        use hyper_util::server::conn::auto;
+
+        tokio::spawn(async move {
+            let mut builder = auto::Builder::new(TokioExecutor::new());
+            // http/1 settings
+            builder.http1().timer(TokioTimer::new());
```

Comment:
What's this doing?  Do we not need something similar for HTTP2?

Reply:
This tells hyper how to use a timer for any timeouts (like a header read timeout). There isn't currently a similar method for `http2()`.

Reply:
Out of curiosity, is that because HTTP2 doesn't need similar timeouts for some reason?

Reply:
Maybe. But as of right now, the builder doesn't have such a method.

Reply:
@seanmonstar I was looking at this code again for other reasons and seeing if I could commonize the HTTP and HTTPS paths a bit more.  When we talked about this above you said there wasn't a similar method for HTTP2, but in the non-TLS path, there seems to be one and we use it:
https://github.com/seanmonstar/dropshot/blob/35c897b26aa0f9241c5b523a84ec009e8999ba97/dropshot/src/server.rs#L274-L275

Is there a reason these paths should be different for TLS vs. non-TLS?

Reply:
Huh, I have no idea what I was thinking. I clearly used it elsewhere... I checked blame on the `hyper-util` repo, in case it was a brand new addition after my saying that... nope. I'm at a complete loss.

There's no reason it should be different for TLS vs not.

Reply:
No worries!  Thanks for the quick reply.  I'll fix this in #1121.